agent/tool-scripts/pprof

```diff
@@ -101,88 +102,75 @@ done
-                if [ $(systemctl list-unit-files | grep atomic-openshift-master.service | awk {'print $2'}) == "masked" ]; then
-                    if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_master_api; then
-                        systemctl restart atomic-openshift-master-api atomic-openshift-master-controllers
-                    else
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master_api
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master_controllers
-                        echo "OPENSHIFT_PROFILE_PORT=6061" >> $ose_openshift_master_api
-                        systemctl restart atomic-openshift-master-api.service atomic-openshift-master-controllers.service
-                    fi
-                else
-                    if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_master; then
-                        systemctl restart atomic-openshift-master
-                    else
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master
-                        systemctl restart atomic-openshift-master.service
-                    fi
-                fi
-            fi 
-            ;;
-        node)
-            if [ -e $origin-node ]; then 
-                if grep -q "^OPENSHIFT_PROFILE=web" $origin_node; then
-                    systemctl restart openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $origin_node
-                    systemctl restart openshift-node.service
-                fi 
-            else
-                if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_node; then
-                    systemctl restart atomic-openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_node
-                    systemctl restart atomic-openshift-node.service
-                fi
-            fi 
-    esac
+get_masters() {
+    declare -A master_pids
+    while true; do
+        i=1
+        for master in "${masters[@]}"; do
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${master}:8443 > ${master}_api_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_profile.txt /tmp/${master}_api_profile &
+            master_pids[${master}_api]=$!
+            oc get --raw /debug/pprof/heap --server https://${master}:8443 > ${master}_api_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt /tmp/${master}_api_heap &
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${master}:8444 > ${master}_controller_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_profile.txt /tmp/${master}_controller_profile &
```

Comment:
Shouldn't the `${master}_controller_profile` be in `/tmp`?

Reply:
Yep

Reply:
Corrected in https://github.com/distributed-system-analysis/pbench/pull/847